src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -840,6 +846,57 @@ void WindowImplX11::setSize(const Vector2u& size)
     XFlush(m_display);
 }
 
+////////////////////////////////////////////////////////////
+void WindowImplX11::setMinimumSize(const Vector2u &minimumSize)
+{
+    assert(minimumSize.x < m_maximumSize.x);
+    assert(minimumSize.y < m_maximumSize.y);
```

Comment:
Should this be `<=`?

```suggestion
    assert(minimumSize.x <= m_maximumSize.x);
    assert(minimumSize.y <= m_maximumSize.y);
```

Ditto for the use of `assert`. It means "this should never happen". Do we want these conditions to be documented undefined behavior, or do we want to return a `bool` instead to signal failure?

Reply:
no. If the minimum size and maximum size are the same, that is still a valid state. It's basically forcing the window not to resize. in the respective dimension.